src/main/java/bio/terra/service/filedata/FileMetadataUtils.java

```diff
@@ -109,4 +110,25 @@ public static Set<String> findNewDirectoryPaths(
     }
     return pathsToCheck;
   }
+
+  /**
+   * Given an absolute path to a file or directory, return a List of absolute paths for all parent
+   * directories
+   *
+   * @param path An absolute path
+   * @return An ordered list of paths that could potentially be created. Note: this excludes the
+   *     passed in path
+   */
+  public static List<String> extractDirectoryPaths(String path) {
+    List<String> allPaths = new ArrayList<>();
+    String interimPath = "";
+    allPaths.add("/");
+    for (String part : getDirectoryPath(path).split("/")) {
+      if (!StringUtils.isEmpty(part)) {
+        interimPath += "/" + part;
+        allPaths.add(interimPath);
+      }
+    }
+    return allPaths;
+  }
```

Comment:
This seems a little wasteful as it uses `split` but then re-allocs each prefix string separately. Using `indexOf()` instead of `split` would avoid that duplication.
```
  public static List<String> extractDirectoryPaths2(String path) {
    List<String> allPaths = new ArrayList<>();
    allPaths.add("/");
    int pos = 1;
    while ((pos = path.indexOf("/", pos)) != -1) {
      allPaths.add(path.substring(0, pos++));
    }
    return allPaths;
  }
```

Reply:
Yeah, I just found this way a touch easier to read